Data/sync/Sync.swift

```diff
@@ -459,12 +463,20 @@ extension Sync {
         guard var fetchedRecords = recordType.fetchedModelType?.syncRecords(recordJSON) else { return }
         
         // Currently only prefs are device related
-        if recordType == .prefs, let data = fetchedRecords as? [SyncDevice] {
+        if recordType == .prefs {
             // Devices have really bad data filtering, so need to manually process more of it
             // Sort to not rely on API - Reverse sort, so unique pulls the `latest` not just the `first`
-            fetchedRecords = data.sorted { device1, device2 in
+            fetchedRecords = fetchedRecords.sorted { device1, device2 in
                 device1.syncTimestamp ?? -1 > device2.syncTimestamp ?? -1 }.unique { $0.objectId ?? [] == $1.objectId ?? [] }
             
+        } else if recordType == .bookmark {
+            // Bookmars are sorted to have the oldest bookmark to come in first, so it can be added before it's children.
```

Comment:
How is this changing current behavior?

Reply:
Currently bookmarks from the server are received in random order I believe, but if we want to add them properly we have to sort it.
Otherwise nested bookmark could go in before its parent which resulted in children bookmarks being outside of their parent folder.

Reply:
I see, this should be handled when a folder is added though, when a folder is added we already attempt to find all orphaned children.
https://github.com/brave/brave-ios/pull/454/files#diff-01f57e7d7609dba0eec131782b05f57dR210

Reply:
But if we first save children and parent next, the children will stay at root level. It doesn't remove the Bookmark and move it into a correct place.
The link you got me doesn't take me to any particular line btw

Reply:
Alright, we'll take care of this separately.